proto/osmosis/valset-pref/v1beta1/tx.proto

```diff
@@ -98,6 +105,27 @@ message MsgUndelegateFromValidatorSet {
 
 message MsgUndelegateFromValidatorSetResponse {}
 
+message MsgUndelegateFromRebalancedValidatorSet {
+  option (amino.name) = "osmosis/MsgUndelegateFromRebalancedValidatorSet";
+
+  // delegator is the user who is trying to undelegate.
+  string delegator = 1 [ (gogoproto.moretags) = "yaml:\"delegator\"" ];
+
+  // the amount the user wants to undelegate
+  // For ex: Undelegate 50 osmo with validator-set {ValA -> 0.5, ValB -> 0.5}
+  // Our undelegate logic would first check the current delegation balance.
+  // If the user has 90 osmo delegated to ValA and 10 osmo delegated to ValB,
+  // the rebalanced validator set would be {ValA -> 0.9, ValB -> 0.1}
+  // So now the 45 osmo would be undelegated from ValA and 5 osmo would be
+  // undelegated from ValB.
+  cosmos.base.v1beta1.Coin coin = 3 [
```

Comment:
```suggestion
  cosmos.base.v1beta1.Coin coin = 2 [
```

Reply:
Assuming this was unintentional since this is a new message

Reply:
Correct, bumped the number down https://github.com/osmosis-labs/osmosis/pull/6685/commits/948db2bb0370be31af9a2cc5bcf029a350c188e2